### PR TITLE
Fixed #30647 -- Fixed crash of autoreloader when extra directory cannot be resolved.

### PR DESCRIPTION
**Pull request by orf**
Ticket: https://code.djangoproject.com/ticket/30647

### CODE REVIEW

File: django/utils/autoreload.py [hunk 1]
```diff
@@ -236,15 +236,25 @@ def __init__(self):
...
         path = Path(path)
-        if not path.is_absolute():
-            raise ValueError('%s must be absolute.' % path)
+
+        try:
+            path = path.absolute()
+        except FileNotFoundError:
+            logger.debug('Skipping watching file %s as it cannot be resolved.', path, exc_info=True)
+            return
+
```
Comment (felixxm) [inline django/utils/autoreload.py]: Do we need these changes? :thinking: `Path.absolute()` raises `FileNotFoundError`, so why not catch it in `watch_for_translation_changes()`, e.g.
```python
for path in directories:
    try:
        absolute_path = path.absolute()
        sender.watch_dir(absolute_path, '**/*.mo')
    except FileNotFoundError:
        logger.debug('Skipping watching file %s as it cannot be resolved.', path, exc_info=True)

```

Comment (orf) [reply django/utils/autoreload.py]: I felt like this logic should belong in the reloader as it's a bit easier to test, and keeps it all centralized. I'm happy to do it like you suggest if you have strong feelings.

Comment (felixxm) [reply django/utils/autoreload.py]: I'm just afraid that we're changing a current behavior that works properly :thinking:, but I'm fine with your proposition. 

Off-topic: Do we need `watch_file()` at all? It looks unused :thinking: .

Comment (felixxm) [reply django/utils/autoreload.py]: @orf What do you think about removing `watch_file()`?

Comment (felixxm) [reply django/utils/autoreload.py]: See #11590

Comment (orf) [reply django/utils/autoreload.py]: Sorry for the late reply, yep, lets get rid of it 👍

Comment (felixxm) [reply django/utils/autoreload.py]: Great, thanks! I will merge #11590 and rebase this PR :+1:

**Comment by felixxm** [comment]
@orf Thanks for this patch :+1: I rebased from master, pushed minor edits, and changed `test_watch_dir_with_unresolvable_path` to use `watch_dir()` (previously both tests use `watch_file()`).

**Comment by liampauling** [comment]
Thanks for fixing this so quickly! 

Just tested locally off the 2.2 stable branch and no issues.

**Pull request merged**